src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Semantic/SemanticTokenTestBase.cs

```diff
@@ -96,40 +98,70 @@ protected string GetBaselineFileContents(string baselineFileName)
         string documentText, Range razorRange, bool isRazorFile, int hostDocumentSyncVersion = 0)
     {
         var codeDocument = CreateCodeDocument(documentText, isRazorFile, DefaultTagHelpers);
-        var csharpRange = GetMappedCSharpRange(codeDocument, razorRange);
-        var csharpTokens = Array.Empty<int>();
+        int[][]? csharpTokens = null;
+        var csharpDocumentUri = new Uri("C:\\TestSolution\\TestProject\\TestDocument.cs");
+        var csharpSourceText = codeDocument.GetCSharpSourceText();
 
-        if (csharpRange is not null)
-        {
-            var csharpDocumentUri = new Uri("C:\\TestSolution\\TestProject\\TestDocument.cs");
-            var csharpSourceText = codeDocument.GetCSharpSourceText();
+        await using var csharpServer = await CSharpTestLspServerHelpers.CreateCSharpLspServerAsync(
+            csharpSourceText, csharpDocumentUri, SemanticTokensServerCapabilities, SpanMappingService, DisposalToken);
 
-            await using var csharpServer = await CSharpTestLspServerHelpers.CreateCSharpLspServerAsync(
-                csharpSourceText, csharpDocumentUri, SemanticTokensServerCapabilities, SpanMappingService, DisposalToken);
-            var result = await csharpServer.ExecuteRequestAsync<SemanticTokensRangeParams, SemanticTokens>(
-                Methods.TextDocumentSemanticTokensRangeName,
-                CreateVSSemanticTokensRangeParams(csharpRange, csharpDocumentUri),
-                DisposalToken);
+        var csharpRanges = GetMappedCSharpRanges(codeDocument, razorRange);
+        if (csharpRanges is not null)
+        {
+            SemanticTokens[]? responses = null;
+            if (UsePreciseSemanticTokenRanges)
+            {
+                responses = new SemanticTokens[csharpRanges.Length];
+                for (var i = 0; i < csharpRanges.Length; i++)
+                {
+                    var result = await csharpServer.ExecuteRequestAsync<SemanticTokensRangeParams, SemanticTokens>(
+                        Methods.TextDocumentSemanticTokensRangeName,
+                        CreateVSSemanticTokensRangeParams(csharpRanges[i], csharpDocumentUri),
+                        DisposalToken);
+
+                    responses[i] = result;
+                }
+            }
+            else
+            {
+                responses = new SemanticTokens[1];
+                var result = await csharpServer.ExecuteRequestAsync<SemanticTokensRangeParams, SemanticTokens>(
+                    Methods.TextDocumentSemanticTokensRangeName,
+                    CreateVSSemanticTokensRangeParams(csharpRanges[0], csharpDocumentUri),
```

Comment:
>csharpRanges[0] [](http://example.com/codeflow?start=54&length=15)

Did you want to change this code to more closely model the changes you made to SemanticToken.ProvideSemanticTokensRangesAsync?